### PR TITLE
feat(button): add link, disabled, secondary, primary design implement…

### PR DESCRIPTION
**Pull request by RifianFernando**
feat(button): add link, disabled, secondary, primary design implement…

Developer: Rifian F.

### Task Summary

1. Create button link and fixing colorized
2. create button primary, secondary, link, for disabled buttton

### CODE REVIEW

**Comment by randiadiel** [comment]
Will review ya, dont merge

**Review by randiadiel — CHANGES_REQUESTED** [review]
Nitip Sendal

File: src/components/Button/styles.ts [hunk 1]
```diff
@@ -1,49 +1,88 @@
 import { css } from '@emotion/css';
 import ColorSystem from '../ColorSystem/index';
+import { generateDefaultPrimarySecondaryButtonStyles } from './utils/StylesGenerator';
```
Comment (randiadiel) [inline src/components/Button/styles.ts:3]: Sebenernya kalo utils namanya ga PascalCase. Tapi camelCase

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -20,15 +20,18 @@ const Button = (props: PropsWithChildren<ButtonProps>): JSX.Element => {
...
         children,
         ...rest
     } = props;
-    const className = cssClassName(btnSize, btnType);
+    const className = disabled
+        ? `${cssClassName(btnSize, btnType)} disabled`
+        : cssClassName(btnSize, btnType);
 
-    const color = { backgroundColor: btnColor };
+    const color = { backgroundColor: btnColor, border: btnColor };
```
[the remark this answers is not in the document]
Comment (RifianFernando) [reply src/components/Button/index.tsx:34]: Oke ko done ko

File: src/components/Button/styles.ts (hunk 1 above)
Comment (RifianFernando) [reply src/components/Button/styles.ts:3]: oke ko udh aku coba pake camelCase nya thankyou ko sarannya

Comment (RifianFernando) [reply src/components/Button/styles.ts:3]: kalo nama functionnya gini oke gk ko kira kira namingnya?

File: src/components/Button/styles.ts [hunk 3]
```diff
@@ -1,49 +1,88 @@
...
+                color: ColorSystem.GREY300,
+                cursor: 'not-allowed',
+            },
         },
     }),
 
     Link: css({
-        border: 'none',
-        backgroundColor: 'transparent',
+        border: 'none !important',
```
[the remark this answers is not in the document]
Comment (RifianFernando) [reply src/components/Button/styles.ts:73]: oke ko aman ko udh aku ganti ko

File: src/components/Button/index.tsx [hunk 4]
```diff
@@ -20,15 +24,18 @@ const Button = (props: PropsWithChildren<ButtonProps>): JSX.Element => {
...
         children,
         ...rest
     } = props;
-    const className = cssClassName(btnSize, btnType);
+    const className = disabled
+        ? `${cssClassName(btnSize, btnType, btnColor)} disabled`
+        : cssClassName(btnSize, btnType, btnColor);
 
-    const color = { backgroundColor: btnColor };
+    // const styleProperties = { backgroundColor: btnColor, border: btnColor };
```
Comment (randiadiel) [inline src/components/Button/index.tsx:38]: Hmmmmmm

File: src/components/Button/index.tsx [hunk 5]
```diff
@@ -50,8 +57,8 @@ const Button = (props: PropsWithChildren<ButtonProps>): JSX.Element => {
             <button
                 {...rest}
                 onClick={handleClick}
-                style={color}
                 className={className}
+                // style={styleProperties}
```
Comment (randiadiel) [inline src/components/Button/index.tsx:61]: Hmmmmm

**Review by randiadiel — APPROVED** [review]
Dibenerin ya comment"nya

Besides that lgtm

File: src/components/Button/index.tsx (hunk 4 above)
Comment (RifianFernando) [reply src/components/Button/index.tsx:38]: done ko

File: src/components/Button/index.tsx (hunk 5 above)
Comment (RifianFernando) [reply src/components/Button/index.tsx:61]: done ko

**Pull request merged**